tableau_blueprints/download_view.py

```diff
@@ -0,0 +1,157 @@
+        req_option = TSC.RequestOptions()
+        req_option.filter.add(TSC.Filter(TSC.RequestOptions.Field.Name,
+                                         TSC.RequestOptions.Operator.Equals,
+                                         workbook_name))
+        with connection:
+            all_workbooks = server.workbooks.get(req_options=req_option)
+            workbook_id= [workbook.id for workbook in all_workbooks[0] if workbook.name == workbook_name]
+            if workbook_id[0] is not None:
+                # get the workbook item
+                workbook = server.workbooks.get_by_id(workbook_id[0])
+
+                # get the view information
+                server.workbooks.populate_views(workbook)
+
+                # print information about the views for the work item
+                view_info=[view.id for view in workbook.views if view.name == view_name]
+            else:
+                print(f'{view_name} could not be found for the given workbook {workbook_name}.'
+                      f'Please check for typos and ensure that the name you provide matches exactly (case senstive)')
+                sys.exit(EXIT_CODE_INVALID_RESOURCE)
+
+    except Exception as e:
+        print(f'{view_name} could not be found for the given workbook {workbook_name}.'
+              f'Please check for typos and ensure that the name you provide matches exactly (case senstive)')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_RESOURCE)
+    if len(view_info)<=0:
+        print(f'{view_name} could not be found or your user does not have access. '
+              f'Please check for typos and ensure that the name you provide matches exactly (case senstive)')
+        sys.exit(EXIT_CODE_INVALID_RESOURCE)
+    else:
+        view_id=view_info[0]
+    return view_id
+
+
+def download_view_item(server, connection, filename, filetype, view_id, view_name):
+    """to download a view from Tableau Server
+
+    :param server:
+    :param connection: Tableau connection object
+    :param filename: name of the view to be downloaded as.
+    :param filetype: 'image', 'thumbnail', 'pdf', 'csv'
+    :return:
+    """
+    file_path = os.path.dirname(os.path.realpath(__file__))
+    try:
+        with connection:
+            views = server.views.get_by_id(view_id)
+            if filetype == "image":
+                server.views.populate_image(views)
+                with open('./' + filename, 'wb') as f:
+                    f.write(views.image)
+
+            if filetype == "pdf":  # Populate and save the PDF data in a file
+                server.views.populate_pdf(views, req_options=None)
+                with open('./' + filename, 'wb') as f:
+                    f.write(views.pdf)
+
+            if filetype == "csv":
+                server.views.populate_csv(views, req_options=None)
+                with open('./' + filename, 'wb') as f:
+                    # Perform byte join on the CSV data
+                    f.write(b''.join(views.csv))
```

Comment:
What data type is `views.csv`? Is joining that type with an empty string appropriate, or is a more sophisticated CSV writing required?